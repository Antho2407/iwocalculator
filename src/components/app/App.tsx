import React, { FC, useState } from "react";

import { AppRoot, Title } from "./App.components";
import TopMenu from "../topMenu/TopMenu";
import Product from "../product/Product";
import { RCF, BL } from "../product/calculation";

const App: FC = () => {
  const [amount, setAmount] = useState(10000);
  const [duration, setDuration] = useState(4);

  return (
    <AppRoot>
      <Title>Your loan</Title>
      <TopMenu
        onAmountChanged={setAmount}
        onDurationChanged={setDuration}
        amount={amount}
        duration={duration}
      />
      {amount && duration && (
        <>
          <Product
            label="Resolving Credit Facility"
            calculation={RCF}
            amount={amount}
            duration={duration}
          />
          <Product
            label="Business loan"
            calculation={BL}
            amount={amount}
            duration={duration}
          />
        </>
      )}
    </AppRoot>
  );
};

export default App;
