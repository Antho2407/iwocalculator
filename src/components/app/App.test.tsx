import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import App from "./App";

describe("<App />", () => {
  let wrapper: ShallowWrapper;

  describe("When the component is mounted", () => {
    it("should render successfully", () => {
      wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
