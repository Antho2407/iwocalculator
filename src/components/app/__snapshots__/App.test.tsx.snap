// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> When the component is mounted should render successfully 1`] = `
<styled.div>
  <styled.h1>
    Your loan
  </styled.h1>
  <TopMenu
    amount={10000}
    duration={4}
    onAmountChanged={[Function]}
    onDurationChanged={[Function]}
  />
  <Product
    amount={10000}
    calculation={[Function]}
    duration={4}
    label="Resolving Credit Facility"
  />
  <Product
    amount={10000}
    calculation={[Function]}
    duration={4}
    label="Business loan"
  />
</styled.div>
`;
